timApp/util/flask/search.py

```diff
@@ -189,220 +178,164 @@ def to_json(self):
-        return {"results": results, "num_results": self.get_match_count()}
+@dataclass()
+class TagResult(SearchResult):
+    """
+    Document tag search result
+    """
 
-    def get_match_count(self) -> int:
-        """
-        :return: How many match words the title has.
-        """
-        if len(self.word_results) > 0:
-            return len(self.word_results)
-        else:
-            return self.alt_num_results
+    pass
+
+
+def get_search_match_count(results: list[SearchResult], rt: Type[SearchResult]) -> int:
+    """
+    Returns the total number of matches found for a specific type of search
+    :return: total number of matches found for the search
+    """
+    if results and not type(results[0]) is rt:
+        results = list(result for result in results if result.__class__ is rt)
+    return sum(r.get_match_count() for r in results)
 
 
+@dataclass()
 class DocResult:
     """
     Contains one document's title and word search information.
     """
 
-    def __init__(
-        self, doc_info: DocInfo, par_results=None, title_results=None, incomplete=False
-    ):
-        if par_results is None:
-            par_results = []
-        if title_results is None:
-            title_results = []
-        self.doc_info = doc_info
-        self.par_results = par_results
-        self.title_results = title_results
-        self.incomplete = incomplete
-
-    def add_par_result(self, result: ParResult) -> None:
-        """
-        Add new paragraph search result to the list.
-
-        :param result: New paragraph result.
-        :return: None.
-        """
-        self.par_results.append(result)
+    doc_info: DocInfo
+    incomplete: bool = False
+    search_results: list[SearchResult] = field(default_factory=list)
 
-    def add_title_result(self, result: TitleResult) -> None:
+    def add_search_result(self, result: SearchResult) -> None:
         """
-        Add new title search result to the list.
-
-        :param result: New title result.
-        :return: None.
+        Add a new search result to document results
+        :param result: Search result to add
+        :return: None
         """
-        self.title_results.append(result)
+        self.search_results.append(result)
 
     def has_results(self) -> bool:
         """
         :return: Whether the document has any results in it.
         """
-        return len(self.par_results) > 0 or len(self.title_results) > 0
+        return len(self.search_results) > 0
 
     def to_json(self):
         """
         :return: A dictionary of the object, suitable for JSON-conversion.
         """
-        par_result_dicts = []
-        for r in self.par_results:
-            par_result_dicts.append(r)
-        title_result_dicts = []
-        for r in self.title_results:
-            title_result_dicts.append(r)
+        title_results = list(
+            title_res
+            for title_res in self.search_results
+            if title_res.__class__ is TitleResult
```

Comment:
Mielellään "pythonimaisempi"

```suggestion
            if isinstance(title_res, TitleResult)
```

ja vastaavasti alemmilla